src/main/java/Controller/MainController.java

```diff
@@ -2,109 +2,101 @@
+                            setText(notificationGroup.getName());
+                        }
+                    }
+
+                };
+                return cell;
+            }
 
-    private void loadTreeItems() {
+        });
+        listView.setOnMouseClicked(new EventHandler<MouseEvent>() {
+            @Override
+            public void handle(MouseEvent event) {
+                Group selected =(Group)listView.getSelectionModel().getSelectedItem();
+                NotificationGroupButtonClicked(selected);
+            }
+        });
+    }
 
-        dashboard = new TreeItem<>("Dashboard");
-        createNotification = new TreeItem<>("Create Notification");
-        settings = new TreeItem<>("Settings");
 
-        groupRoot = new TreeItem<String>("Groups");
-        groupRoot.setExpanded(false);
+    private void NotificationGroupButtonClicked(Group group) {
+        System.out.println(group.getName() + " Group button clicked");
 
-        root = new TreeItem<String>("Root Node");
-        root.getChildren().addAll(dashboard, createNotification, groupRoot, settings);
-        locationTreeView.setRoot(root);
-        locationTreeView.setShowRoot(false);
     }
 
     private ArrayList<String> getGroupsNames(ArrayList<Group> groups) {
         ArrayList<String> groupNameList = new ArrayList<>();
         for (Group group : groups) {
-            groupNameList.add(group.getId());
+            groupNameList.add(group.getName());
         }
         return groupNameList;
     }
 
     private void getGroups() {
-        ArrayList<Group> groupArrayList = new ArrayList<>();
+        groupArrayList = new ArrayList<>();
         final FirebaseDatabase database = FirebaseDatabase.getInstance();
         DatabaseReference ref = database.getReference().child("group");
         ref.addValueEventListener(new ValueEventListener() {
             @Override
             public void onDataChange(DataSnapshot snapshot) {
-                groupRoot.getChildren().clear();
+                groupArrayList.clear();
                 for (DataSnapshot singleGroup: snapshot.getChildren()){
-                    Group newGroup = new Group((String)singleGroup.child("id").getValue(),(String)singleGroup.child("name").getValue());
+                    ArrayList<String> workerIdArrayList = new ArrayList();
+                    for(DataSnapshot workerSnapshot : singleGroup.child("members").getChildren()){
+                        workerIdArrayList.add((String)workerSnapshot.getValue());
+                    }
+                    Group newGroup = new Group((String)singleGroup.child("id").getValue(),(String)singleGroup.child("name").getValue(),workerIdArrayList);
                     groupArrayList.add(newGroup);
-                    System.out.println(newGroup.getName());
-                    groupRoot.getChildren().add(new TreeItem<>(newGroup.getName()));
+                    //System.out.println(newGroup.getName());
```

Comment:
Should this commented out line be removed?

Reply:
Yes, I'll remove them. they were just for debugging.